modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -446,6 +477,75 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
   generateMetadata();
 }
 
+void
+PinMeshGenerator::generateFlexibleAssemblyBoundaries()
```

Comment:
Can this method and the one in AssemblyMG be merged?

Reply:
They behave slightly different in terms of defining the variables `outermost_block_name` and `has_single_mesh_interval`, and which subgenerators are called from this method. It'd be easier to implement as separate methods, merging them would require more input parameters to be passed into a common method

Reply:
okay